### ISSUE TITLE
Re-enable one-click generation of new passwords

### ISSUE THREAD

**Issue by skibbipl** [kee-org/browser-addon#68]
Firefox: 58.0a1 (2017-11-11) (64 bits)
Kee: 2.0.0.21b3
KeePass: 2.37
KeePass RPC: 1.7.3

When I select generate new password in Kee, popup window says that the password is copied to clipboard - but in fact it's not. However if I select manually some profile (for example random MAC key) it is copied correctly to clipboard.
There are no errors in console:
```
In browser content page script, received message from background script
Logging system initialised at Sun Nov 12 2017 13:08:02 GMT+0100
iframe page starting
Logging system config updated at Sun Nov 12 2017 13:08:02 GMT+0100
iframe page ready
In iframe script, received message from background script: 
In browser content page script, received message from background script
```

**Comment by luckyrat** [comment]
Thanks for reporting the issue.

Everything within Kee is working as expected except that the call to `document.execCommand("copy");` does not actually put the new password onto the clipboard.

This is only a problem in Firefox and only when triggered from the popup window - context menu and keyboard shortcuts work fine.

**Comment by luckyrat** [comment]
It sounds as though there will be new ways to do this in future versions of Firefox - https://bugzilla.mozilla.org/show_bug.cgi?id=1272869 & https://bugzilla.mozilla.org/show_bug.cgi?id=1344410

In the mean time, because it is not feasible to detect when this Firefox bug will actually prevent the clipboard operation from succeeding, I will have to disable the one-click password generation feature and require all users to make the extra selection of a password profile every time.

Unfortunately, the only other alternative would be to leave this mixed behaviour that varies based on browser and invocation method. Since this feature could be utilised in a way that could lead to data loss if a bug is present, we just can't take the risk of misleading users.

**Comment by luckyrat** [comment]
This is worked around as described, in beta 6: https://addons.mozilla.org/en-US/firefox/addon/keefox/versions/2.0.0.24b6

**Comment by asteres** [comment]
It would be great to have some sort of user feedback, I wasn't aware that was doing anything and kept pressing generate password, later finding them all in my database.

**Comment by light** [comment]
@luckyrat fyi the one-click generation feature is not disabled, it does not copy the password to the clipboard but a new password is still generated in KeePass.

**Comment by luckyrat** [comment]
@light Thanks. My main aim with the last-minute workaround was avoiding the incorrect information being presented to the user but it's wasteful to create backups of passwords that can never be used so the next release will disable the generation part as well as the clipboard part that's already disabled.

Mozilla have shown no sign of prioritising the changes that could let us re-enable the old functionality so I'm going to mark this as blocked for the foreseeable future.

**Comment by luckyrat** [comment]
The linked bugs are now resolved (e.g. https://blog.mozilla.org/addons/2018/08/31/extensions-in-firefox-63/) but before unblocking this I want to wait to see what comes of the Chrome manifest v3 planning - talk there indicates the removal of background pages which would cripple a lot of things, and clipboard access may be included in that.

Will revisit all these questions once Firefox ESR is upgraded to a version that supports that new clipboard API (late 2019).

**Comment by luckyrat** [comment]
With the changes to the interface in recent Kee versions (including the soon-to-be-released 3.6) and ongoing problems with clipboard access bugs in both Chrome and Firefox, I no longer feel this is a feature worth keeping on the roadmap.
